eos-paygd/main.c

```diff
@@ -35,8 +35,90 @@
 #include <sys/un.h>
 #include <libeos-payg/efi.h>
 
+#define LOGFILE_DIRNAME "/var/log/eos-payg"
+#define LOGFILE_BASENAME "eos-paygd"
+#define LOGFILE_EXT "log"
+
 #define TIMEOUT_POWEROFF_ON_ERROR_MINUTES 20
 
+static GIOChannel *
+open_log_file (void)
+{
+  const gchar *log_file_name = NULL;
+  g_autoptr(GFile) log_file = NULL;
+  g_autoptr(GDateTime) now = NULL;
+  g_autofree gchar *tstamp = NULL;
+
+  /* Build log file name with a date stamp so we get one file per day,
+   * ex., eos-paygd-20220418.log
+   */
+  now = g_date_time_new_now_local ();
+  tstamp = g_date_time_format (now, "%Y%m%d");
+  log_file_name = g_strconcat (LOGFILE_DIRNAME, "/", LOGFILE_BASENAME, "-",
```

Comment:
Technically this should be `g_build_filename()` rather than `g_strconcat()`, for portability to other platforms and for the case where the file system encoding is not UTF-8 (which is possible but vanishingly rare). Just an fyi, though, this isn’t worth changing now. The filename would have to be constructed separately (using `g_strconcat()`) before building the path.

Reply:
Actually, the return value from `g_strconcat()` is leaked here, so this does need some tweaking.